src/main/java/com/vizor/unreal/util/Misc.java

```diff
@@ -246,6 +246,71 @@ else if (numBrackets < 0)
         }
     }
 
+    // s6fix @bernst - Generate C++ safe and compilable code.
+
+    /**
+     * Converts a string like "com.s6.core.image-service" to ComS6CoreImageService.
+     * Preserves case, except to upper case first letter to a word, where words are separated by underscores.
+     *
+     * In the future we can consider reworking the code to transform ssomething like "com.s6.core.image-service to "ComS6Core_ImageService".
+     * 
+     * @return Input "com.s6.core.image-service" to "ComS6CoreImageService"
+     */
+    public static String mixedCharacterStringToCppSafe(final String mixedCharacterString)
+    {
+        // Transforms input "com.s6.core.image-service" to "com_s6_core_image_service"
+        final String cppSafeString = mixedCharacterString.replaceAll("[.-]", "_");
+        
+        // Below here, transforms input "com_s6_core_image_service" to "ComS6CoreImageService"
+
+        // snakeCaseToCamelCase but modified to leave existing case alone.
+        // Some strings that come in here are mostly preformated, leave the case alone.
+        final StringBuilder sb = new StringBuilder(cppSafeString.length());
```

Comment:
Can we make this just call `snakeCaseToCamelCase` and then make the first letter uppercase? Also, can we make that its own method `snakeCaseToPascalCase`?